src/thread.ts

```diff
@@ -0,0 +1,115 @@
+import { StreamChat } from './client';
+import {
+  DefaultGenerics,
+  ExtendableGenerics,
+  MessageResponse,
+  ThreadResponse,
+  ChannelResponse,
+  FormatMessageResponse,
+  ReactionResponse,
+  UserResponse,
+} from './types';
+import { addToMessageList, formatMessage } from './utils';
+
+type ThreadReadStatus<StreamChatGenerics extends ExtendableGenerics = DefaultGenerics> = Record<
```

Comment:
Where is the thread read status updated / based on what WS event?

Reply:
hmm right now I am delegating this part to SDK implementation. But I will think about it for 2nd round, maybe we can handle it by default within js client.